testing/environments/docker/elasticsearch_kerberos/scripts/installkdc.sh

```diff
@@ -67,12 +62,21 @@ EOF
 kadmin.local -q "addprinc -pw elastic admin/admin@$REALM_NAME"
 kadmin.local -q "ktadd -k /etc/admin.keytab admin/admin@$REALM_NAME"
 
-# set ownership for ES
+# set ownership for ES so that we can start krb5kdc in non-root mode
 chown -R elasticsearch:elasticsearch $LOGDIR
 chown -R elasticsearch:elasticsearch $KDC_CONFIG
+
+## ensure we can access database principle
+chown -R elasticsearch:elasticsearch /var/lib/krb5kdc/
+chown -R elasticsearch:elasticsearch /var/lib/krb5kdc/principal 
+
+## stash is where the master password is stored
+chown -R elasticsearch:elasticsearch $LOCALSTATEDIR/krb5kdc/
+chown -R elasticsearch:elasticsearch $LOCALSTATEDIR/krb5kdc/stash
+
 chown -R elasticsearch:elasticsearch $LOCALSTATEDIR/krb5.conf
+chown -R elasticsearch:elasticsearch $LOCALSTATEDIR/krb5kdc/kadm5.acl
 chown -R elasticsearch:elasticsearch $LOCALSTATEDIR/admin.keytab
 
-
 # Create a link so addprinc.sh is on path
-ln -s /scripts/addprinc.sh /usr/bin/
+sudo ln -s /scripts/addprinc.sh /usr/bin/
```

Comment:
```suggestion
ln -s /scripts/addprinc.sh /usr/bin/
```

It looks like the whole file is executed with root permissions so sudo seems redundant here